<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The DiffSegR R package takes the bam files from an RNA-Seq dataset with two biological conditions and returns transcriptome-wide expression differences between these two conditions without using pre-existing annotations (typically genes).">
<title>DiffSegR: An RNA-Seq Data Driven Method for Differential Expression Analysis using Changepoint Detection • DiffSegR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="DiffSegR: An RNA-Seq Data Driven Method for Differential Expression Analysis using Changepoint Detection">
<meta property="og:description" content="The DiffSegR R package takes the bam files from an RNA-Seq dataset with two biological conditions and returns transcriptome-wide expression differences between these two conditions without using pre-existing annotations (typically genes).">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">DiffSegR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/introductory_tutorial.html">Introductory tutorial</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/advanced_uses.html">Advanced tutorials</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aLiehrmann/DiffSegR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="diffsegr-">DiffSegR <a class="anchor" aria-label="anchor" href="#diffsegr-"></a>
</h1>
</div>
<div class="section level2">
<h2 id="what-is-diffsegr-">What is DiffSegR ?<a class="anchor" aria-label="anchor" href="#what-is-diffsegr-"></a>
</h2>
<p>The <code>DiffSegR</code> R package takes the bam files from an RNA-Seq dataset with two biological conditions and returns transcriptome-wide expression differences between these two conditions without using pre-existing annotations (typically genes).</p>
<p>All the motivations for this method and a comparison with the state-of-the-art are described in Liehrmann et al. (2023). Briefly, gaining a global picture of gene and RNA regulations requires detailed knowledge of the transcriptome along with the enzymatic and RNA-binding activities that shape it. Hence, several RNA-Seq-based strategies have recently been developed to decipher its complexity. However, most of the tools developed only count the abundance of sequencing reads along annotated regions. These annotations are typically incomplete and often lead to errors in the differential expression analysis.</p>
<p>In the context of differential expression analysis, the regulations of genes and RNAs result in local changes of the log2 fold change (log2-FC) along the genome, i.e the differential transcription profile. Therefore, one can use a multiple changepoint detection algorithm such as <code>fpop</code> (Maidstone et al. 2017) to delineate the boundaries of the differentially expressed regions (DERs) without relying on pre-existing annotations. The DERs can then be assessed using, for example, the negative binomial generalized linear model of <code>DESeq2</code> (Love et al. 2014).</p>
<p>The <code>DiffSegR</code> R package, an identify then annotate tool, implements this strategy.</p>
</div>
<div class="section level2">
<h2 id="how-can-i-get-diffsegr-">How can I get DiffSegR ?<a class="anchor" aria-label="anchor" href="#how-can-i-get-diffsegr-"></a>
</h2>
<p>Make sure that <code>remotes</code> is installed by running <code>install.packages("remotes")</code>, then type</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"aLiehrmann/DiffSegR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="where-can-i-learn-more">Where can I learn more?<a class="anchor" aria-label="anchor" href="#where-can-i-learn-more"></a>
</h2>
<p>See the <a href="articles/introductory_tutorial.html">introductory tutorial</a> for an introduction to our framework on real data. See the <a href="articles/advanced_uses.html">advanced tutorial</a> for examples of more advanced uses of <code>DiffSegR</code>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Liehrmann, A., Delannoy, E., Castandet, B. and Rigaill, G. DiffSegR: an RNA-Seq data driven method for differential expression analysis using changepoint detection (2023). <a href="">add link to preprint</a>.</p>
<p>Maidstone R, Hocking T, Rigaill G, Fearnhead P. On optimal multiple changepoint algorithms for large data. Stat Comput 27(2), 519-533 (2017). <a href="https://link.springer.com/article/10.1007/s11222-016-9636-3" class="external-link">doi:10.1007/s11222-016-9636-3</a>.</p>
<p>Love, M.I., Huber, W. &amp; Anders, S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol 15, 550 (2014). <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8" class="external-link">doi:10.1186/s13059-014-0550-8</a>.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/aLiehrmann/DiffSegR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/aLiehrmann/DiffSegR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing DiffSegR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Arnaud Liehrmann <br><small class="roles"> Author, maintainer </small>  </li>
<li>Benoît Castandet <br><small class="roles"> Author </small>  </li>
<li>Guillem Rigaill <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Arnaud Liehrmann, Benoît Castandet, Guillem Rigaill.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
