<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="DiffSegR">
<title>Tutorial : Transcriptome-Wide Differential Expression Analysis with DiffSegR • DiffSegR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial : Transcriptome-Wide Differential Expression Analysis with DiffSegR">
<meta property="og:description" content="DiffSegR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DiffSegR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/introductory_tutorial.html">Introductory tutorial</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/advanced_uses.html">Advanced tutorials</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aLiehrmann/DiffSegR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Tutorial : Transcriptome-Wide Differential Expression Analysis with DiffSegR</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/aLiehrmann/DiffSegR/blob/HEAD/vignettes/introductory_tutorial.Rmd" class="external-link"><code>vignettes/introductory_tutorial.Rmd</code></a></small>
      <div class="d-none name"><code>introductory_tutorial.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>In this tutorial we use an RNA-Seq dataset (SRA046998) generated to study the differences in Arabidopsis thaliana chloroplast RNA metabolism between a control (wt) and a null mutant of the chloroplast-targeted <span class="math inline">\(3'\to5'\)</span> exoribonuclease polynucleotide phosphorylase (pnp1-1). In comparison to control plants, this mutant overaccumulate RNA fragments that are mainly located outside of the annotated genic areas and the RNA-Seq data have been extensively validated using molecular techniques (Castandet et al. 2013; Hotto et al. 2015). Both conditions contain two biological replicates. Only a diluted set of reads overlapping positions 71950 to 78500 on chloroplast genome has been kept as a minimal dataset example.</p>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>After setting up the necessary working environnement and loading the software, we are going through each step of a classical differential expression analysis along the genome with <code>DiffSegR</code>:</p>
<ol style="list-style-type: decimal">
<li>Computation of coverages and per-base <span class="math inline">\(\log_2\)</span>-FC from bam files ;</li>
<li>Summarize the differential transcription landscape, i.e. (i) identify segments along the genome in the per-base <span class="math inline">\(\log_2\)</span>-FC and (ii) count reads overlapping them ;</li>
<li>Use the counts as measures of the expression abundance of the segments for differential expression analysis ;</li>
<li>Annotate the differential expressed regions (DERs) ;</li>
<li>Visualize the DERs.</li>
</ol>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>First, we create a working directory that will host all the files produced by <code>DiffSegR</code> during the analysis (coverages, results).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">working_directory</span> <span class="op">&lt;-</span> <span class="st">"./DIFFSEGR_TEST"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">working_directory</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Then, we save sample information that we will use for this tutorial in a text file. In practice, one can create the working directory and write the text file by hand. The sample information file needs to include the following columns:</p>
<ol style="list-style-type: decimal">
<li>sample : identifier for the sample ;</li>
<li>condition : identifier for the condition ;</li>
<li>replicate : identifier for the replicate ;</li>
<li>bam : path to the bam file (.bam) ;</li>
<li>coverage : path to the coverage file (.rds).</li>
</ol>
<p>Note: Coverages are computed internally by <code>DiffSegR</code> from bam files and saved in <em>rds</em> format at the location specified by the user (<code>coverage</code>).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#- create sample information table --------------------------------------------#</span></span>
<span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  sample    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>,</span>
<span>  condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1"</span>, <span class="st">"wt"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>  replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span>
<span>  bam       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1_ChrC_71950_78500.bam"</span>, </span>
<span>      <span class="st">"pnp1_1_2_ChrC_71950_78500.bam"</span>,</span>
<span>      <span class="st">"wt_1_ChrC_71950_78500.bam"</span>,</span>
<span>      <span class="st">"wt_2_ChrC_71950_78500.bam"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    <span class="kw">function</span><span class="op">(</span><span class="va">bam</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">bam</span>, package <span class="op">=</span> <span class="st">"DiffSegR"</span><span class="op">)</span></span>
<span>  <span class="op">)</span>,</span>
<span>  coverage  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span>
<span>    <span class="va">working_directory</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>, <span class="st">".rds"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#- save sample information table ----------------------------------------------# </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span></span>
<span>  <span class="va">sample_info</span>, </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>, <span class="st">"sample_info.txt"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#- display sample information table -------------------------------------------#</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">sample_info</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="6%">
<col width="6%">
<col width="6%">
<col width="60%">
<col width="19%">
</colgroup>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">condition</th>
<th align="right">replicate</th>
<th align="left">bam</th>
<th align="left">coverage</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">pnp1_1_1</td>
<td align="left">pnp1_1</td>
<td align="right">1</td>
<td align="left">/tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/pnp1_1_1_ChrC_71950_78500.bam</td>
<td align="left">./DIFFSEGR_TEST/pnp1_1_1.rds</td>
</tr>
<tr class="even">
<td align="left">pnp1_1_2</td>
<td align="left">pnp1_1</td>
<td align="right">2</td>
<td align="left">/tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/pnp1_1_2_ChrC_71950_78500.bam</td>
<td align="left">./DIFFSEGR_TEST/pnp1_1_2.rds</td>
</tr>
<tr class="odd">
<td align="left">wt_1</td>
<td align="left">wt</td>
<td align="right">1</td>
<td align="left">/tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/wt_1_ChrC_71950_78500.bam</td>
<td align="left">./DIFFSEGR_TEST/wt_1.rds</td>
</tr>
<tr class="even">
<td align="left">wt_2</td>
<td align="left">wt</td>
<td align="right">2</td>
<td align="left">/tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/wt_2_ChrC_71950_78500.bam</td>
<td align="left">./DIFFSEGR_TEST/wt_2.rds</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="analysis">Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<p>Now, we present a basic use of the main functions of the <code>DiffSegR</code> R package. The analysis was carried on <code>4</code> logical cores. This parameter can be modified by the user according to the resources at his disposal.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/aLiehrmann/DiffSegR" class="external-link">DiffSegR</a></span><span class="op">)</span></span>
<span><span class="va">nb_threads</span> <span class="op">=</span> <span class="fl">4</span></span></code></pre></div>
<div class="section level3">
<h3 id="step-1-computation-of-coverages-and-per-base-log_2-fc-from-bam-files">Step 1: Computation of coverages and per-base <span class="math inline">\(\log_2\)</span>-FC from bam files<a class="anchor" aria-label="anchor" href="#step-1-computation-of-coverages-and-per-base-log_2-fc-from-bam-files"></a>
</h3>
<p><code><a href="../reference/loadData.html">loadData()</a></code>:</p>
<ol style="list-style-type: decimal">
<li>builds coverages from bam files within a locus specified by the user. Here we focus on positions <code>71950</code> to <code>78500</code> of chloroplast genome (<code>ChrC</code>). If the reads are stranded (<code>stranded = TRUE</code>), the function builds one coverage profile per-strand for each replicate of both compared biological conditions. By default the heuristic used to compute coverages is the geometric mean of 5’ and 3’ profiles of the aligned reads (<code>coverageType = "average"</code>) ;</li>
<li>transforms coverages into the per-base <span class="math inline">\(\log_2\)</span>-FC (one by strand). The reference biological condition specified by the user is set to the denominator (<code>referenceCondition = "wt"</code>) ;</li>
<li>returns the coverages and the per-base <span class="math inline">\(\log_2\)</span>-FC as a list of run-length encoding objects.</li>
</ol>
<p>Note: If coverages have already been computed, set <code>fromBam = FALSE</code> to load them directly from specified <em>rds</em> files.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadData.html">loadData</a></span><span class="op">(</span></span>
<span>  sampleInfo   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>,<span class="st">"sample_info.txt"</span><span class="op">)</span>,</span>
<span>  locus        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    seqid      <span class="op">=</span> <span class="st">"ChrC"</span>, </span>
<span>    chromStart <span class="op">=</span> <span class="fl">71950</span>, </span>
<span>    chromEnd   <span class="op">=</span> <span class="fl">78500</span></span>
<span>  <span class="op">)</span>,</span>
<span>  referenceCondition <span class="op">=</span> <span class="st">"wt"</span>,</span>
<span>  stranded           <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  fromBam            <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  nbThreads          <span class="op">=</span> <span class="va">nb_threads</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; building coverage profiles ...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         ==========     _____ _    _ ____  _____  ______          _____  </span></span>
<span><span class="co">#&gt;         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span></span>
<span><span class="co">#&gt;           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span></span>
<span><span class="co">#&gt;             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span></span>
<span><span class="co">#&gt;               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span></span>
<span><span class="co">#&gt;         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span></span>
<span><span class="co">#&gt;        Rsubread 2.10.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; //========================== featureCounts setting ===========================\\</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||             Input files : 4 BAM files                                      ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span></span>
<span><span class="co">#&gt; ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span></span>
<span><span class="co">#&gt; ||                           wt_1_ChrC_71950_78500.bam                        ||</span></span>
<span><span class="co">#&gt; ||                           wt_2_ChrC_71950_78500.bam                        ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||              Paired-end : no                                               ||</span></span>
<span><span class="co">#&gt; ||        Count read pairs : no                                               ||</span></span>
<span><span class="co">#&gt; ||              Annotation : R data.frame                                     ||</span></span>
<span><span class="co">#&gt; ||      Dir for temp files : .                                                ||</span></span>
<span><span class="co">#&gt; ||                 Threads : 4                                                ||</span></span>
<span><span class="co">#&gt; ||                   Level : meta-feature level                               ||</span></span>
<span><span class="co">#&gt; ||      Multimapping reads : counted                                          ||</span></span>
<span><span class="co">#&gt; || Multi-overlapping reads : not counted                                      ||</span></span>
<span><span class="co">#&gt; ||   Min overlapping bases : 1                                                ||</span></span>
<span><span class="co">#&gt; ||          Read reduction : to 3' end                                        ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; \\============================================================================//</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; //================================= Running ==================================\\</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Load annotation file .Rsubread_UserProvidedAnnotation_pid1360629 ...       ||</span></span>
<span><span class="co">#&gt; ||    Features : 13102                                                        ||</span></span>
<span><span class="co">#&gt; ||    Meta-features : 13102                                                   ||</span></span>
<span><span class="co">#&gt; ||    Chromosomes/contigs : 1                                                 ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 24260                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 24249 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 21100                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 21098 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 17533                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 17531 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 14070                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 14066 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Write the final count table.                                               ||</span></span>
<span><span class="co">#&gt; || Write the read assignment summary.                                         ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; \\============================================================================//</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         ==========     _____ _    _ ____  _____  ______          _____  </span></span>
<span><span class="co">#&gt;         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span></span>
<span><span class="co">#&gt;           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span></span>
<span><span class="co">#&gt;             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span></span>
<span><span class="co">#&gt;               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span></span>
<span><span class="co">#&gt;         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span></span>
<span><span class="co">#&gt;        Rsubread 2.10.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; //========================== featureCounts setting ===========================\\</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||             Input files : 4 BAM files                                      ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span></span>
<span><span class="co">#&gt; ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span></span>
<span><span class="co">#&gt; ||                           wt_1_ChrC_71950_78500.bam                        ||</span></span>
<span><span class="co">#&gt; ||                           wt_2_ChrC_71950_78500.bam                        ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||              Paired-end : no                                               ||</span></span>
<span><span class="co">#&gt; ||        Count read pairs : no                                               ||</span></span>
<span><span class="co">#&gt; ||              Annotation : R data.frame                                     ||</span></span>
<span><span class="co">#&gt; ||      Dir for temp files : .                                                ||</span></span>
<span><span class="co">#&gt; ||                 Threads : 4                                                ||</span></span>
<span><span class="co">#&gt; ||                   Level : meta-feature level                               ||</span></span>
<span><span class="co">#&gt; ||      Multimapping reads : counted                                          ||</span></span>
<span><span class="co">#&gt; || Multi-overlapping reads : not counted                                      ||</span></span>
<span><span class="co">#&gt; ||   Min overlapping bases : 1                                                ||</span></span>
<span><span class="co">#&gt; ||          Read reduction : to 5' end                                        ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; \\============================================================================//</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; //================================= Running ==================================\\</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Load annotation file .Rsubread_UserProvidedAnnotation_pid1360629 ...       ||</span></span>
<span><span class="co">#&gt; ||    Features : 13102                                                        ||</span></span>
<span><span class="co">#&gt; ||    Meta-features : 13102                                                   ||</span></span>
<span><span class="co">#&gt; ||    Chromosomes/contigs : 1                                                 ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 24260                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 24251 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 21100                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 21088 (99.9%)                        ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 17533                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 17520 (99.9%)                        ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 14070                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 14060 (99.9%)                        ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Write the final count table.                                               ||</span></span>
<span><span class="co">#&gt; || Write the read assignment summary.                                         ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; \\============================================================================//</span></span>
<span><span class="co">#&gt; building log2-FC per-base ...</span></span>
<span><span class="co">#&gt; finished in 1.35417079925537 secs</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="co">#&gt; $coverages</span></span>
<span><span class="co">#&gt; $coverages$plus</span></span>
<span><span class="co">#&gt; $coverages$plus$pnp1_1_1</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 5255 runs</span></span>
<span><span class="co">#&gt;   Lengths:       86        1        3        1 ...        1        1       10</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$plus$pnp1_1_2</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 5209 runs</span></span>
<span><span class="co">#&gt;   Lengths:       89        1        2        1 ...        5        2       30</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.732051 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$plus$wt_1</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 4610 runs</span></span>
<span><span class="co">#&gt;   Lengths:       46        1       15        1 ...       23        1       14</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$plus$wt_2</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 4601 runs</span></span>
<span><span class="co">#&gt;   Lengths:       57        1       26        1 ...       22        1        2</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$minus</span></span>
<span><span class="co">#&gt; $coverages$minus$pnp1_1_1</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 526 runs</span></span>
<span><span class="co">#&gt;   Lengths:       31        1       68        1 ...        6        1        1</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$minus$pnp1_1_2</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 585 runs</span></span>
<span><span class="co">#&gt;   Lengths:      406        1       42        1 ...        2        1        1</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$minus$wt_1</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 463 runs</span></span>
<span><span class="co">#&gt;   Lengths:      427        1       83        1 ...        4        1        2</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $coverages$minus$wt_2</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 544 runs</span></span>
<span><span class="co">#&gt;   Lengths:      458        1       63        1 ...        1       18        1</span></span>
<span><span class="co">#&gt;   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.414214 0.000000 0.732051</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $log2FoldChange</span></span>
<span><span class="co">#&gt; $log2FoldChange$plus</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 6004 runs</span></span>
<span><span class="co">#&gt;   Lengths:         46          1         10 ...          1          2</span></span>
<span><span class="co">#&gt;   Values :  0.0000000 -0.2500000  0.0000000 ...  -0.250000   0.000000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $log2FoldChange$minus</span></span>
<span><span class="co">#&gt; numeric-Rle of length 6551 with 1239 runs</span></span>
<span><span class="co">#&gt;   Lengths:           31            1           68 ...            1            1</span></span>
<span><span class="co">#&gt;   Values :         0.00         0.25         0.00 ...  5.00000e-01 -3.96241e-01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $locus</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames      ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]     ChrC 71950-78500      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stranded</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $sampleInfo</span></span>
<span><span class="co">#&gt;                                 sample condition replicate</span></span>
<span><span class="co">#&gt; pnp1_1_1_ChrC_71950_78500.bam pnp1_1_1    pnp1_1         1</span></span>
<span><span class="co">#&gt; pnp1_1_2_ChrC_71950_78500.bam pnp1_1_2    pnp1_1         2</span></span>
<span><span class="co">#&gt; wt_1_ChrC_71950_78500.bam         wt_1        wt         1</span></span>
<span><span class="co">#&gt; wt_2_ChrC_71950_78500.bam         wt_2        wt         2</span></span>
<span><span class="co">#&gt;                                                                                                                     bam</span></span>
<span><span class="co">#&gt; pnp1_1_1_ChrC_71950_78500.bam /tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/pnp1_1_1_ChrC_71950_78500.bam</span></span>
<span><span class="co">#&gt; pnp1_1_2_ChrC_71950_78500.bam /tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/pnp1_1_2_ChrC_71950_78500.bam</span></span>
<span><span class="co">#&gt; wt_1_ChrC_71950_78500.bam         /tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/wt_1_ChrC_71950_78500.bam</span></span>
<span><span class="co">#&gt; wt_2_ChrC_71950_78500.bam         /tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/wt_2_ChrC_71950_78500.bam</span></span>
<span><span class="co">#&gt;                                                   coverage</span></span>
<span><span class="co">#&gt; pnp1_1_1_ChrC_71950_78500.bam ./DIFFSEGR_TEST/pnp1_1_1.rds</span></span>
<span><span class="co">#&gt; pnp1_1_2_ChrC_71950_78500.bam ./DIFFSEGR_TEST/pnp1_1_2.rds</span></span>
<span><span class="co">#&gt; wt_1_ChrC_71950_78500.bam         ./DIFFSEGR_TEST/wt_1.rds</span></span>
<span><span class="co">#&gt; wt_2_ChrC_71950_78500.bam         ./DIFFSEGR_TEST/wt_2.rds</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $referenceCondition</span></span>
<span><span class="co">#&gt; [1] "wt"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="setp-2-summarizing-the-differential-transcription-landscape">Setp 2: Summarizing the differential transcription landscape<a class="anchor" aria-label="anchor" href="#setp-2-summarizing-the-differential-transcription-landscape"></a>
</h3>
<p><code><a href="../reference/segmentation.html">segmentation()</a></code>:</p>
<ol style="list-style-type: decimal">
<li>calls <code><a href="https://rdrr.io/pkg/fpopw/man/Fpop_w.html" class="external-link">fpopw::Fpop_w()</a></code>, a changepoint detection in the mean of a Gaussian signal approach, on the per-base <span class="math inline">\(\log_2\)</span>-FC of both strands to retrieve the segments boundaries (changepoints). The hyperparameter <span class="math inline">\(\alpha\)</span> specified by the user controls the number of returned segments. The number of changepoints is a decreasing function of <span class="math inline">\(\alpha\)</span>. By default <code>alpha=2</code>. The segments are stored as a GenomicRanges object.</li>
<li>calls <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code> which assigns to the segments the mapped reads from each replicate of each biological condition. A read is allowed to be assigned to more than one feature if it is found to overlap with more than one feature. <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code> ends by building a count matrix.</li>
<li>returns the segments and associated count matrix as a <code>SummarizedExperiment</code> object.</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SExp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/segmentation.html">segmentation</a></span><span class="op">(</span></span>
<span>  data                   <span class="op">=</span> <span class="va">data</span>, </span>
<span>  nbThreadsFeatureCounts <span class="op">=</span> <span class="va">nb_threads</span>,</span>
<span>  outputDirectory        <span class="op">=</span> <span class="va">working_directory</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; changepoint detection on ChrC 71950 78500 plus ...</span></span>
<span><span class="co">#&gt;   &gt; model selection ...</span></span>
<span><span class="co">#&gt;   &gt; report available in ./DIFFSEGR_TEST/all_models_plus.pdf</span></span>
<span><span class="co">#&gt; changepoint detection on ChrC 71950 78500 minus ...</span></span>
<span><span class="co">#&gt;   &gt; model selection ...</span></span>
<span><span class="co">#&gt;   &gt; report available in ./DIFFSEGR_TEST/all_models_minus.pdf</span></span>
<span><span class="co">#&gt; finished in 0.605228185653687 secs</span></span>
<span><span class="co">#&gt; counting ...</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         ==========     _____ _    _ ____  _____  ______          _____  </span></span>
<span><span class="co">#&gt;         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span></span>
<span><span class="co">#&gt;           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span></span>
<span><span class="co">#&gt;             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span></span>
<span><span class="co">#&gt;               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span></span>
<span><span class="co">#&gt;         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span></span>
<span><span class="co">#&gt;        Rsubread 2.10.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; //========================== featureCounts setting ===========================\\</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||             Input files : 4 BAM files                                      ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span></span>
<span><span class="co">#&gt; ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span></span>
<span><span class="co">#&gt; ||                           wt_1_ChrC_71950_78500.bam                        ||</span></span>
<span><span class="co">#&gt; ||                           wt_2_ChrC_71950_78500.bam                        ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; ||              Paired-end : no                                               ||</span></span>
<span><span class="co">#&gt; ||        Count read pairs : no                                               ||</span></span>
<span><span class="co">#&gt; ||              Annotation : R data.frame                                     ||</span></span>
<span><span class="co">#&gt; ||      Dir for temp files : .                                                ||</span></span>
<span><span class="co">#&gt; ||                 Threads : 4                                                ||</span></span>
<span><span class="co">#&gt; ||                   Level : meta-feature level                               ||</span></span>
<span><span class="co">#&gt; ||      Multimapping reads : counted                                          ||</span></span>
<span><span class="co">#&gt; || Multi-overlapping reads : counted                                          ||</span></span>
<span><span class="co">#&gt; ||   Min overlapping bases : 1                                                ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; \\============================================================================//</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; //================================= Running ==================================\\</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Load annotation file .Rsubread_UserProvidedAnnotation_pid1360629 ...       ||</span></span>
<span><span class="co">#&gt; ||    Features : 69                                                           ||</span></span>
<span><span class="co">#&gt; ||    Meta-features : 69                                                      ||</span></span>
<span><span class="co">#&gt; ||    Chromosomes/contigs : 1                                                 ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 24260                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 24260 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 21100                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 21100 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 17533                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 17533 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span></span>
<span><span class="co">#&gt; ||    Strand specific : stranded                                              ||</span></span>
<span><span class="co">#&gt; ||    Single-end reads are included.                                          ||</span></span>
<span><span class="co">#&gt; ||    Total alignments : 14070                                                ||</span></span>
<span><span class="co">#&gt; ||    Successfully assigned alignments : 14070 (100.0%)                       ||</span></span>
<span><span class="co">#&gt; ||    Running time : 0.00 minutes                                             ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; || Write the final count table.                                               ||</span></span>
<span><span class="co">#&gt; || Write the read assignment summary.                                         ||</span></span>
<span><span class="co">#&gt; ||                                                                            ||</span></span>
<span><span class="co">#&gt; \\============================================================================//</span></span>
<span><span class="co">#&gt; finished in 0.171851396560669 secs</span></span></code></pre></div>
<p>Between positions <code>71950</code> and <code>78500</code> of chloroplast genome, <code><a href="../reference/segmentation.html">segmentation()</a></code> returns 17 segments on the forward (+) strand and 52 segments on the reverse (-) strand. Segment coordinates and associated counts can be retrieved using respectively <code><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">SummarizedExperiment::rowRanges</a></code> and <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment::assay</a></code> accessors.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#- number of segments found ---------------------------------------------------# </span></span>
<span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu">strand</span><span class="op">(</span><span class="va">SExp</span><span class="op">)</span></span>
<span><span class="co">#&gt; factor-Rle of length 69 with 2 runs</span></span>
<span><span class="co">#&gt;   Lengths: 17 52</span></span>
<span><span class="co">#&gt;   Values :  +  -</span></span>
<span><span class="co">#&gt; Levels(3): + - *</span></span>
<span><span class="co">#- display first five segments ------------------------------------------------#</span></span>
<span><span class="va">segment_coordinates</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">SExp</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">segment_coordinates</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">segment_coordinates</span><span class="op">)</span><span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"seqnames"</span>,</span>
<span>  <span class="st">"start"</span>,</span>
<span>  <span class="st">"end"</span>,</span>
<span>  <span class="st">"width"</span>,</span>
<span>  <span class="st">"strand"</span><span class="op">)</span></span>
<span><span class="op">]</span><span class="op">)</span> </span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">width</th>
<th align="left">strand</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">71950</td>
<td align="right">72255</td>
<td align="right">306</td>
<td align="left">+</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">72256</td>
<td align="right">72402</td>
<td align="right">147</td>
<td align="left">+</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">72403</td>
<td align="right">74038</td>
<td align="right">1636</td>
<td align="left">+</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74039</td>
<td align="right">74305</td>
<td align="right">267</td>
<td align="left">+</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74306</td>
<td align="right">74841</td>
<td align="right">536</td>
<td align="left">+</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#- display counts associated to first five segments ---------------------------#</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">SExp</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">counts</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">pnp1_1_1</th>
<th align="right">pnp1_1_2</th>
<th align="right">wt_1</th>
<th align="right">wt_2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ChrC_71950_72255_plus</td>
<td align="right">143</td>
<td align="right">139</td>
<td align="right">84</td>
<td align="right">84</td>
</tr>
<tr class="even">
<td align="left">ChrC_72256_72402_plus</td>
<td align="right">649</td>
<td align="right">665</td>
<td align="right">326</td>
<td align="right">293</td>
</tr>
<tr class="odd">
<td align="left">ChrC_72403_74038_plus</td>
<td align="right">13203</td>
<td align="right">11731</td>
<td align="right">10849</td>
<td align="right">8366</td>
</tr>
<tr class="even">
<td align="left">ChrC_74039_74305_plus</td>
<td align="right">155</td>
<td align="right">132</td>
<td align="right">98</td>
<td align="right">104</td>
</tr>
<tr class="odd">
<td align="left">ChrC_74306_74841_plus</td>
<td align="right">2337</td>
<td align="right">1996</td>
<td align="right">1589</td>
<td align="right">1218</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="step-3-differential-expression-analysis-dea">Step 3: Differential expression analysis (DEA)<a class="anchor" aria-label="anchor" href="#step-3-differential-expression-analysis-dea"></a>
</h3>
<p><code><a href="../reference/dea.html">dea()</a></code> function takes as input the recently built <code>SummarizedExperiment</code> object and calls <code>DESeq2</code> to test for each line, i.e. each segment, the difference in expression between the two compared biological conditions. Here we control all the resulting p-values using a Benjamini–Hochberg procedure at level <code>significanceLevel = 1e-2</code>.</p>
<p>Note: By default sample-specific size factors are computed by <code>DESeq2</code>. They can be overwritten by the user. In this example, sample-specific size factors have the same weight <code>sizeFactors = rep(1,4)</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dea.html">dea</a></span><span class="op">(</span></span>
<span>  data              <span class="op">=</span> <span class="va">data</span>,</span>
<span>  SExp              <span class="op">=</span> <span class="va">SExp</span>,</span>
<span>  design            <span class="op">=</span> <span class="op">~</span> <span class="va">condition</span>,</span>
<span>  sizeFactors       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span>,</span>
<span>  significanceLevel <span class="op">=</span> <span class="fl">1e-2</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;   &gt; differential expression analysis ...</span></span>
<span><span class="co">#&gt; renaming the first element in assays to 'counts'</span></span>
<span><span class="co">#&gt; gene-wise dispersion estimates</span></span>
<span><span class="co">#&gt; mean-dispersion relationship</span></span>
<span><span class="co">#&gt; final dispersion estimates</span></span>
<span><span class="co">#&gt;   &gt;  multiple testing correction ...</span></span>
<span><span class="co">#&gt;   &gt;  normalize denoised log2-FC and coverage ...</span></span></code></pre></div>
<p>Regions differentially expressed (DERs) are set to <code>rejectedHypotheses = TRUE</code>. On this example, <code><a href="../reference/dea.html">dea()</a></code> returns 39 DERs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#- number of DERs -------------------------------------------------------------#</span></span>
<span><span class="va">DERs</span> <span class="op">&lt;-</span> <span class="va">dds</span><span class="op">[</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">$</span><span class="va">rejectedHypotheses</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">DERs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 39</span></span>
<span><span class="co">#- display first five DERs by position ----------------------------------------#</span></span>
<span><span class="va">DERs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges</a></span><span class="op">(</span><span class="va">DERs</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">DERs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">DERs</span><span class="op">)</span><span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  <span class="st">"seqnames"</span>,</span>
<span>  <span class="st">"start"</span>,</span>
<span>  <span class="st">"end"</span>,</span>
<span>  <span class="st">"width"</span>,</span>
<span>  <span class="st">"strand"</span>,</span>
<span>  <span class="st">"log2FoldChange"</span>,</span>
<span>  <span class="st">"padj"</span><span class="op">)</span></span>
<span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">width</th>
<th align="left">strand</th>
<th align="right">log2FoldChange</th>
<th align="right">padj</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">72256</td>
<td align="right">72402</td>
<td align="right">147</td>
<td align="left">+</td>
<td align="right">1.0859541</td>
<td align="right">0.0000025</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74306</td>
<td align="right">74841</td>
<td align="right">536</td>
<td align="left">+</td>
<td align="right">0.6263374</td>
<td align="right">0.0037268</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74842</td>
<td align="right">74938</td>
<td align="right">97</td>
<td align="left">+</td>
<td align="right">1.0248663</td>
<td align="right">0.0000195</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74939</td>
<td align="right">75471</td>
<td align="right">533</td>
<td align="left">+</td>
<td align="right">1.2142538</td>
<td align="right">0.0000001</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">76882</td>
<td align="right">77054</td>
<td align="right">173</td>
<td align="left">+</td>
<td align="right">1.7448836</td>
<td align="right">0.0000000</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="step-4-annotate-the-ders">Step 4: Annotate the DERs<a class="anchor" aria-label="anchor" href="#step-4-annotate-the-ders"></a>
</h3>
<p><code><a href="../reference/annotateNearest.html">annotateNearest()</a></code> relies on user specified annotations in <em>gff3</em> or <em>gtf</em> format to annotate the DERs found during the DEA. Seven classes of labels translate the relative positions of the DER and its closest annotation(s): antisense; upstream; downstream; inside; overlapping 3’; overlapping 5’; overlapping 5’, 3’.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">exons_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span> <span class="st">"extdata"</span>, </span>
<span> <span class="st">"TAIR10_ChrC_exon.gff3"</span>, </span>
<span> package <span class="op">=</span> <span class="st">"DiffSegR"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">annotated_DERs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/annotateNearest.html">annotateNearest</a></span><span class="op">(</span></span>
<span>   data        <span class="op">=</span> <span class="va">data</span>,</span>
<span>   dds         <span class="op">=</span> <span class="va">dds</span>,</span>
<span>   annotations <span class="op">=</span> <span class="va">exons_path</span>,</span>
<span>   select      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"width"</span>, <span class="st">"strand"</span>, </span>
<span>     <span class="st">"description"</span>, <span class="st">"distance"</span>, <span class="st">"Parent"</span>, <span class="st">"log2FoldChange"</span>, <span class="st">"pvalue"</span><span class="op">)</span>,</span>
<span>   orderBy     <span class="op">=</span> <span class="st">"pvalue"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;  &gt; loading annotations from /tmp/RtmphcvDQk/temp_libpath14b710619d9ca0/DiffSegR/extdata/TAIR10_ChrC_exon.gff3 ...</span></span>
<span><span class="co">#&gt;  &gt; annotate differential expressed regions ...</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">annotated_DERs</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, digits <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">seqnames</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">width</th>
<th align="left">strand</th>
<th align="left">description</th>
<th align="right">distance</th>
<th align="left">Parent</th>
<th align="right">log2FoldChange</th>
<th align="right">pvalue</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">77964</td>
<td align="right">78112</td>
<td align="right">149</td>
<td align="left">+</td>
<td align="left">downstream</td>
<td align="right">291</td>
<td align="left">ATCG00730.1</td>
<td align="right">4.642</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">77964</td>
<td align="right">78112</td>
<td align="right">149</td>
<td align="left">+</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00740.1</td>
<td align="right">4.642</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">76882</td>
<td align="right">77054</td>
<td align="right">173</td>
<td align="left">+</td>
<td align="left">upstream</td>
<td align="right">143</td>
<td align="left">ATCG00730.1</td>
<td align="right">1.745</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">77740</td>
<td align="right">77963</td>
<td align="right">224</td>
<td align="left">+</td>
<td align="left">downstream</td>
<td align="right">67</td>
<td align="left">ATCG00730.1</td>
<td align="right">2.319</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">77740</td>
<td align="right">77963</td>
<td align="right">224</td>
<td align="left">+</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00740.1</td>
<td align="right">2.319</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">77055</td>
<td align="right">77192</td>
<td align="right">138</td>
<td align="left">+</td>
<td align="left">upstream</td>
<td align="right">5</td>
<td align="left">ATCG00730.1</td>
<td align="right">1.929</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74939</td>
<td align="right">75471</td>
<td align="right">533</td>
<td align="left">+</td>
<td align="left">downstream</td>
<td align="right">92</td>
<td align="left">ATCG00720.1</td>
<td align="right">1.214</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">78113</td>
<td align="right">78214</td>
<td align="right">102</td>
<td align="left">+</td>
<td align="left">downstream</td>
<td align="right">440</td>
<td align="left">ATCG00730.1</td>
<td align="right">3.919</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">78113</td>
<td align="right">78214</td>
<td align="right">102</td>
<td align="left">+</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00740.1</td>
<td align="right">3.919</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">72256</td>
<td align="right">72402</td>
<td align="right">147</td>
<td align="left">+</td>
<td align="left">overlapping 5’</td>
<td align="right">0</td>
<td align="left">ATCG00680.1</td>
<td align="right">1.086</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74001</td>
<td align="right">74019</td>
<td align="right">19</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">229</td>
<td align="left">ATCG00700.1</td>
<td align="right">3.807</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74020</td>
<td align="right">74023</td>
<td align="right">4</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">225</td>
<td align="left">ATCG00700.1</td>
<td align="right">3.747</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">73994</td>
<td align="right">74000</td>
<td align="right">7</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">248</td>
<td align="left">ATCG00700.1</td>
<td align="right">3.858</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74281</td>
<td align="right">74283</td>
<td align="right">3</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.820</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74306</td>
<td align="right">74309</td>
<td align="right">4</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-2.022</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74301</td>
<td align="right">74302</td>
<td align="right">2</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.965</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74842</td>
<td align="right">74938</td>
<td align="right">97</td>
<td align="left">+</td>
<td align="left">overlapping 3’</td>
<td align="right">0</td>
<td align="left">ATCG00720.1</td>
<td align="right">1.025</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74024</td>
<td align="right">74035</td>
<td align="right">12</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">213</td>
<td align="left">ATCG00700.1</td>
<td align="right">3.550</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74284</td>
<td align="right">74300</td>
<td align="right">17</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.735</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74303</td>
<td align="right">74304</td>
<td align="right">2</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.920</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74305</td>
<td align="right">74305</td>
<td align="right">1</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.920</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74036</td>
<td align="right">74039</td>
<td align="right">4</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">209</td>
<td align="left">ATCG00700.1</td>
<td align="right">4.585</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74269</td>
<td align="right">74273</td>
<td align="right">5</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.699</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74040</td>
<td align="right">74077</td>
<td align="right">38</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">171</td>
<td align="left">ATCG00700.1</td>
<td align="right">4.544</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74334</td>
<td align="right">74338</td>
<td align="right">5</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-2.121</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74267</td>
<td align="right">74268</td>
<td align="right">2</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.675</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74274</td>
<td align="right">74277</td>
<td align="right">4</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.651</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74278</td>
<td align="right">74280</td>
<td align="right">3</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.651</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74310</td>
<td align="right">74333</td>
<td align="right">24</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.882</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74339</td>
<td align="right">74339</td>
<td align="right">1</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-2.185</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74264</td>
<td align="right">74265</td>
<td align="right">2</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.631</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74340</td>
<td align="right">74347</td>
<td align="right">8</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-2.066</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74266</td>
<td align="right">74266</td>
<td align="right">1</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.585</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74348</td>
<td align="right">74349</td>
<td align="right">2</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-2.452</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74078</td>
<td align="right">74083</td>
<td align="right">6</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">165</td>
<td align="left">ATCG00700.1</td>
<td align="right">4.143</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74078</td>
<td align="right">74083</td>
<td align="right">6</td>
<td align="left">-</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00690.1</td>
<td align="right">4.143</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">73966</td>
<td align="right">73993</td>
<td align="right">28</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">255</td>
<td align="left">ATCG00700.1</td>
<td align="right">3.755</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">73692</td>
<td align="right">73953</td>
<td align="right">262</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">295</td>
<td align="left">ATCG00700.1</td>
<td align="right">2.566</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">73692</td>
<td align="right">73953</td>
<td align="right">262</td>
<td align="left">-</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00680.1</td>
<td align="right">2.566</td>
<td align="right">0.000</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">73954</td>
<td align="right">73965</td>
<td align="right">12</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">283</td>
<td align="left">ATCG00700.1</td>
<td align="right">4.358</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74261</td>
<td align="right">74263</td>
<td align="right">3</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.409</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74084</td>
<td align="right">74205</td>
<td align="right">122</td>
<td align="left">-</td>
<td align="left">downstream</td>
<td align="right">43</td>
<td align="left">ATCG00700.1</td>
<td align="right">4.285</td>
<td align="right">0.001</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74084</td>
<td align="right">74205</td>
<td align="right">122</td>
<td align="left">-</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00690.1</td>
<td align="right">4.285</td>
<td align="right">0.001</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74306</td>
<td align="right">74841</td>
<td align="right">536</td>
<td align="left">+</td>
<td align="left">overlapping 5’, 3’</td>
<td align="right">0</td>
<td align="left">ATCG00710.1</td>
<td align="right">0.626</td>
<td align="right">0.002</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74306</td>
<td align="right">74841</td>
<td align="right">536</td>
<td align="left">+</td>
<td align="left">overlapping 5’</td>
<td align="right">0</td>
<td align="left">ATCG00720.1</td>
<td align="right">0.626</td>
<td align="right">0.002</td>
</tr>
<tr class="even">
<td align="left">ChrC</td>
<td align="right">74306</td>
<td align="right">74841</td>
<td align="right">536</td>
<td align="left">+</td>
<td align="left">antisense</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">0.626</td>
<td align="right">0.002</td>
</tr>
<tr class="odd">
<td align="left">ChrC</td>
<td align="right">74256</td>
<td align="right">74260</td>
<td align="right">5</td>
<td align="left">-</td>
<td align="left">inside</td>
<td align="right">0</td>
<td align="left">ATCG00700.1</td>
<td align="right">-1.301</td>
<td align="right">0.005</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="step-5-visualizing-the-ders">Step 5: Visualizing the DERs<a class="anchor" aria-label="anchor" href="#step-5-visualizing-the-ders"></a>
</h3>
<p><code><a href="../reference/export.html">export()</a></code> saves for both strands DERs, not-DERs, segmentation, coverages and log2-FC per-base in formats readable (<em>bedGraph</em>, <em>gff3</em>) by genome viewers like the Integrative Genome Viewer (IGV) (Robison et al. 2011). For IGV, <code><a href="../reference/export.html">export()</a></code> also creates an IGV sessiong in <em>xml</em> format that allows loading all tracks in one click.</p>
<p>The tracks from top to bottom stand for:</p>
<ol style="list-style-type: decimal">
<li>(log2-Cov plus) the mean of coverages on the log scale for the forward strand in both biological conditions of interest. Here, the blue line and the red line stand for wt and pnp1-1 conditions, respectively ;</li>
<li>(log2-FC plus) the per-base <span class="math inline">\(\log_2\)</span>-FC between pnp1-1 (numerator) and wt (denominator) for the forward strand. The straight horizontal line is the zero indicator. When the per-base <span class="math inline">\(\log_2\)</span>-FC is above or under the zero indicator line, it suggests an up-regulation and an down-regulation in pnp1-1 compare to wt, respectively. Over the per-base <span class="math inline">\(\log_2\)</span>-FC is displayed the changepoints positions (vertical blue lines) and the mean of each segment (horizontal blue lines that connect two changepoints) ;</li>
<li>(DiffSegR plus) the results of the DEA on the forward strand. Up-regulated regions appear in green. Down-regulated regions appear in purple. Not-DERs appear in grey ;</li>
<li>(annotations) The features provided by the user for interpetations.</li>
</ol>
<p>Symmetrically, the rest of the tracks correspond to the same data on the reverse strand.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ChrC_path</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span> <span class="st">"extdata"</span>, </span>
<span> <span class="st">"TAIR10_ChrC.fa"</span>, </span>
<span> package <span class="op">=</span> <span class="st">"DiffSegR"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/export.html">export</a></span><span class="op">(</span></span>
<span> data            <span class="op">=</span> <span class="va">data</span>,</span>
<span> dds             <span class="op">=</span> <span class="va">dds</span>, </span>
<span> outputDirectory <span class="op">=</span> <span class="va">working_directory</span>,</span>
<span> genome          <span class="op">=</span> <span class="va">ChrC_path</span>,</span>
<span> annotation      <span class="op">=</span> <span class="va">exons_path</span>,</span>
<span> select          <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"log2FoldChange"</span>, <span class="st">"pvalue"</span>, <span class="st">"padj"</span><span class="op">)</span></span>
<span><span class="op">)</span> </span>
<span><span class="co">#&gt; Loading required package: GenomicRanges</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span></span>
<span><span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span></span>
<span><span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span></span>
<span><span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span></span>
<span><span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: GenomeInfoDb</span></span></code></pre></div>
<div class="figure">
<img src="igv_snapshot.png" alt="DiffSegR output between positions 76,990 to 78,248 on the chloroplast genome. The session is loaded in IGV 2.12.3 for Linux." width="100%"><p class="caption">
DiffSegR output between positions 76,990 to 78,248 on the chloroplast genome. The session is loaded in IGV 2.12.3 for Linux.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Castandet B, Hotto AM, Fei Z, Stern DB. Strand-specific RNA sequencing uncovers chloroplast ribonuclease functions. FEBS Lett 587(18), 3096-101 (2013). <a href="https://febs.onlinelibrary.wiley.com/doi/10.1016/j.febslet.2013.08.004" class="external-link">doi:10.1016/j.febslet.2013.08.004</a>.</p>
<p>Hotto AM, Castandet B, Gilet L, Higdon A, Condon C, Stern DB. Arabidopsis chloroplast mini-ribonuclease III participates in rRNA maturation and intron recycling. Plant Cell 27(3), 724-40 (2015). <a href="https://academic.oup.com/plcell/article/27/3/724/6118143" class="external-link">doi:10.1105/tpc.114.134452</a>.</p>
<p>James T. Robinson, Helga Thorvaldsdóttir, Wendy Winckler, Mitchell Guttman, Eric S. Lander, Gad Getz, Jill P. Mesirov. Integrative Genomics Viewer. Nature Biotechnology 29, 24–26 (2011). <a href="https://www.nature.com/articles/nbt.1754" class="external-link">doi:10.1038/nbt.1754</a>.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Arnaud Liehrmann, Benoît Castandet, Guillem Rigaill.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
