<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="dea takes as input the recently built SummarizedExperiment object and calls
DESeq2 to test for each line, i.e. each region, the difference in expression
between the two compared biological conditions. The user can choose to control
all the resulting p-values or only a subset of interest. In the first case
a Benjamini–Hochberg procedure is used to control the False Discovery Rate
(FDR). In the second case, the user specifies selection criteria for the
regions, e.g. a threshold on the absolute log2-FC per-regions. Then, a
post-hoc procedure is used to control the joint Family wise error Rate
(jFWER) on the selected subset of regions."><title>Differential expression analysis (DEA) — dea • DiffSegR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential expression analysis (DEA) — dea"><meta property="og:description" content="dea takes as input the recently built SummarizedExperiment object and calls
DESeq2 to test for each line, i.e. each region, the difference in expression
between the two compared biological conditions. The user can choose to control
all the resulting p-values or only a subset of interest. In the first case
a Benjamini–Hochberg procedure is used to control the False Discovery Rate
(FDR). In the second case, the user specifies selection criteria for the
regions, e.g. a threshold on the absolute log2-FC per-regions. Then, a
post-hoc procedure is used to control the joint Family wise error Rate
(jFWER) on the selected subset of regions."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DiffSegR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/introductory_tutorial.html">Introductory tutorial</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/advanced_uses.html">Advanced tutorials</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Differential expression analysis (DEA)</h1>
      
      <div class="d-none name"><code>dea.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>dea</code> takes as input the recently built <code>SummarizedExperiment</code> object and calls
<code>DESeq2</code> to test for each line, i.e. each region, the difference in expression
between the two compared biological conditions. The user can choose to control
all the resulting p-values or only a subset of interest. In the first case
a Benjamini–Hochberg procedure is used to control the False Discovery Rate
(FDR). In the second case, the user specifies selection criteria for the
regions, e.g. a threshold on the absolute log2-FC per-regions. Then, a
post-hoc procedure is used to control the joint Family wise error Rate
(jFWER) on the selected subset of regions.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">dea</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">SExp</span>,</span>
<span>  design <span class="op">=</span> <span class="op">~</span><span class="va">condition</span>,</span>
<span>  sizeFactors <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  significanceLevel <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  predicate <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  postHoc_significanceLevel <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  postHoc_tdpLowerBound <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  orderBy <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dichotomicSearch <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>data</dt>
<dd><p>The <code>List</code> object returned by <code><a href="loadData.html">loadData()</a></code>.</p></dd>


<dt>SExp</dt>
<dd><p>The <code>SummarizedExperiment</code> object returned by
<code><a href="segmentation.html">segmentation()</a></code>.</p></dd>


<dt>design</dt>
<dd><p>A <code>formula</code> object or a <code>Matrix</code> object. Passed on to
<a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeq2::DESeqDataSet</a>.</p></dd>


<dt>sizeFactors</dt>
<dd><p>A vector of <code>Double</code>. Sample-specific size factors.</p></dd>


<dt>significanceLevel</dt>
<dd><p>A <code>Double</code>. The significance cutoff on q-values.</p></dd>


<dt>predicate</dt>
<dd><p>A predicate function that returns a single TRUE or FALSE if
the region on which it is applied meets the conditions defined in the
predicate. The criteria used in the predicate have to be defined for each
regions in mcols(SExp).</p></dd>


<dt>postHoc_significanceLevel</dt>
<dd><p>A <code>Double</code>. The significance level of the
test procedure (see <a href="https://rdrr.io/pkg/sanssouci/man/posthocBySimes.html" class="external-link">sanssouci::posthocBySimes</a>).</p></dd>


<dt>postHoc_tdpLowerBound</dt>
<dd><p>A <code>Double</code>. The minimum true positive
proportion on the returned set of rejected regions.</p></dd>


<dt>orderBy</dt>
<dd><p>A <code>String</code>. The parameter on which regions are sorted before
the post-hoc procedure.</p></dd>


<dt>verbose</dt>
<dd><p>A <code>Boolean</code>. Should all the operations performed be displayed
(only post-hoc) ?</p></dd>


<dt>dichotomicSearch</dt>
<dd><p>A <code>Boolean</code>. Use it to speed up the post-hoc procedure.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A<code>DESeqDataSet</code> object.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create a working directory for running the example.</span></span></span>
<span class="r-in"><span><span class="va">working_directory</span> <span class="op">&lt;-</span> <span class="st">"./DIFFSEGR_TEST"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">working_directory</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Save sample information in a text file.</span></span></span>
<span class="r-in"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sample    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1"</span>, <span class="st">"wt"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  bam       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1_ChrC_71950_78500.bam"</span>, </span></span>
<span class="r-in"><span>      <span class="st">"pnp1_1_2_ChrC_71950_78500.bam"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"wt_1_ChrC_71950_78500.bam"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"wt_2_ChrC_71950_78500.bam"</span></span></span>
<span class="r-in"><span>     <span class="op">)</span>,</span></span>
<span class="r-in"><span>     <span class="kw">function</span><span class="op">(</span><span class="va">bam</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">bam</span>, package <span class="op">=</span> <span class="st">"DiffSegR"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  coverage  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">working_directory</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>, <span class="st">".rds"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">sample_info</span>, </span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>, <span class="st">"sample_info.txt"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Build coverages and log2-FC per-base.</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadData.html">loadData</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sampleInfo         <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>,<span class="st">"sample_info.txt"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  locus        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqid      <span class="op">=</span> <span class="st">"ChrC"</span>, </span></span>
<span class="r-in"><span>    chromStart <span class="op">=</span> <span class="fl">71950</span>, </span></span>
<span class="r-in"><span>    chromEnd   <span class="op">=</span> <span class="fl">78500</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  referenceCondition <span class="op">=</span> <span class="st">"wt"</span>,</span></span>
<span class="r-in"><span>  stranded           <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  fromBam            <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  nbThreads          <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> building coverage profiles ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : not counted                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Read reduction : to 3' end                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid1296978 ...       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 13102                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 13102                                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24249 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21098 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17531 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14066 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : not counted                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Read reduction : to 5' end                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid1296978 ...       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 13102                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 13102                                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24251 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21088 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17520 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14060 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> building log2-FC per-base ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.486620426177979 secs</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Summarize the differential landscape.</span></span></span>
<span class="r-in"><span><span class="va">SExp</span> <span class="op">&lt;-</span> <span class="fu"><a href="segmentation.html">segmentation</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data                   <span class="op">=</span> <span class="va">data</span>, </span></span>
<span class="r-in"><span>  nbThreadsFeatureCounts <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  outputDirectory        <span class="op">=</span> <span class="va">working_directory</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> changepoint detection on ChrC 71950 78500 plus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; model selection ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; report available in ./DIFFSEGR_TEST/all_models_plus.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> changepoint detection on ChrC 71950 78500 minus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; model selection ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; report available in ./DIFFSEGR_TEST/all_models_minus.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.181750535964966 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> counting ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid1296978 ...       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 69                                                           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 69                                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24260 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21100 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17533 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14070 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.125626564025879 secs</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Differential expression analysis. We control all the resulting </span></span></span>
<span class="r-in"><span><span class="co"># p-values using a Benjamini–Hochberg procedure at level 0.01.</span></span></span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">dea</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data              <span class="op">=</span> <span class="va">data</span>,</span></span>
<span class="r-in"><span>  SExp              <span class="op">=</span> <span class="va">SExp</span>,</span></span>
<span class="r-in"><span>  design            <span class="op">=</span> <span class="op">~</span> <span class="va">condition</span>,</span></span>
<span class="r-in"><span>  sizeFactors       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">4</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  significanceLevel <span class="op">=</span> <span class="fl">1e-2</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; differential expression analysis ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> renaming the first element in assays to 'counts'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt;  multiple testing correction ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt;  normalize denoised log2-FC and coverage ...</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># first to fifth regions </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu">mcols</span><span class="op">(</span><span class="va">dds</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 5 rows and 37 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       modelStart  modelEnd modelMean             featureId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;           &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus          1       306 0.0257894 ChrC_71950_72255_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus        307       453 0.6352520 ChrC_72256_72402_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus        454      2089 0.2826332 ChrC_72403_74038_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus       2090      2356 0.0910074 ChrC_74039_74305_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus       2357      2892 0.3207993 ChrC_74306_74841_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        baseMean    baseVar   allZero dispGeneEst dispGeneIter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;numeric&gt;  &lt;numeric&gt; &lt;logical&gt;   &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus    112.50    1085.67     FALSE  0.00000001            5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus    483.25   40476.25     FALSE  0.00000001            2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus  11037.25 4114258.92     FALSE  0.01999925            2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus    122.25     696.25     FALSE  0.00000001           14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus   1785.00  236376.67     FALSE  0.02248815            2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         dispFit dispersion  dispIter dispOutlier   dispMAP</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;numeric&gt;  &lt;numeric&gt; &lt;integer&gt;   &lt;logical&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus 0.0438822  0.0359269         6       FALSE 0.0359269</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus 0.0230099  0.0189383         6       FALSE 0.0189383</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus 0.0169537  0.0175717         9       FALSE 0.0175717</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus 0.0417124  0.0352381         6       FALSE 0.0352381</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus 0.0183910  0.0192489         9       FALSE 0.0192489</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       Intercept condition_pnp1_1_vs_wt SE_Intercept</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;numeric&gt;              &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus   6.39232               0.747234     0.223109</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus   8.27380               1.085954     0.151892</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus  13.22995               0.375882     0.135628</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus   6.65821               0.506696     0.216738</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus  10.45481               0.626337     0.144130</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       SE_condition_pnp1_1_vs_wt WaldStatistic_Intercept</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                       &lt;numeric&gt;               &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus                  0.307485                 28.6511</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus                  0.210628                 54.4714</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus                  0.191742                 97.5460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus                  0.301495                 30.7200</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus                  0.203189                 72.5374</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       WaldStatistic_condition_pnp1_1_vs_wt WaldPvalue_Intercept</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                  &lt;numeric&gt;            &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus                              2.43015         1.55327e-180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus                              5.15579          0.00000e+00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus                              1.96035          0.00000e+00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus                              1.68061         3.07377e-207</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus                              3.08253          0.00000e+00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       WaldPvalue_condition_pnp1_1_vs_wt  betaConv  betaIter</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                               &lt;numeric&gt; &lt;logical&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus                       1.50926e-02      TRUE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus                       2.52561e-07      TRUE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus                       4.99545e-02      TRUE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus                       9.28388e-02      TRUE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus                       2.05246e-03      TRUE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        deviance  maxCooks  baseMean log2FoldChange     lfcSE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;numeric&gt; &lt;logical&gt; &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus   32.5318        NA    112.50       0.747234  0.307485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus   41.1243        NA    483.25       1.085954  0.210628</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus   67.8832        NA  11037.25       0.375882  0.191742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus   33.5075        NA    122.25       0.506696  0.301495</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus   53.7430        NA   1785.00       0.626337  0.203189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                            stat      pvalue        padj log2FoldChangeMean</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;          &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus   2.43015 1.50926e-02 2.36679e-02          0.0257894</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus   5.15579 2.52561e-07 2.48953e-06          0.6352520</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus   1.96035 4.99545e-02 7.33374e-02          0.2826332</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus   1.68061 9.28388e-02 1.23190e-01          0.0910074</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus   3.08253 2.05246e-03 3.72683e-03          0.3207993</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       log2RefMean log2OtherMean rejectedHypotheses     group</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                         &lt;numeric&gt;     &lt;numeric&gt;          &lt;logical&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus    0.185676      0.211465              FALSE         0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus    0.877737      1.512989               TRUE         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus    1.849847      2.132480              FALSE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus    0.192321      0.283328              FALSE         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus    1.037875      1.358675               TRUE         3</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># delete working directory </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">working_directory</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Arnaud Liehrmann, Benoît Castandet, Guillem Rigaill.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

