<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="loadData() loads :
the sample information related to an RNA-seq experiment ;
the coverage profile overlapping the locus specified by the user
(computed from bam files) ;
the log2-FC per-base (computed from coverages).

"><title>Compute the differential transcription profile (log2-FC per-base) — loadData • DiffSegR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Compute the differential transcription profile (log2-FC per-base) — loadData"><meta property="og:description" content="loadData() loads :
the sample information related to an RNA-seq experiment ;
the coverage profile overlapping the locus specified by the user
(computed from bam files) ;
the log2-FC per-base (computed from coverages).

"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DiffSegR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/introductory_tutorial.html">Introductory tutorial</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/advanced_uses.html">Advanced tutorials</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aLiehrmann/DiffSegR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Compute the differential transcription profile (log2-FC per-base)</h1>
      <small class="dont-index">Source: <a href="https://github.com/aLiehrmann/DiffSegR/blob/HEAD/R/loadData.r" class="external-link"><code>R/loadData.r</code></a></small>
      <div class="d-none name"><code>loadData.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>loadData()</code> loads :</p><ul><li><p>the sample information related to an RNA-seq experiment ;</p></li>
<li><p>the coverage profile overlapping the locus specified by the user
(computed from bam files) ;</p></li>
<li><p>the log2-FC per-base (computed from coverages).</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">loadData</span><span class="op">(</span></span>
<span>  <span class="va">sampleInfo</span>,</span>
<span>  <span class="va">referenceCondition</span>,</span>
<span>  <span class="va">locus</span>,</span>
<span>  stranded <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  strandSpecific <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  fromBam <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  coverageType <span class="op">=</span> <span class="st">"average"</span>,</span>
<span>  readLength <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  isPairedEnd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nbThreads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  featureCountsOtherParams <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>sampleInfo</dt>
<dd><p>A <code>String</code>. Path to the file with
sample information. The file includes the following columns:</p><ul><li><p>sample    : identifier for the sample ;</p></li>
<li><p>condition : identifier for the condition ;</p></li>
<li><p>replicate : identifier for the replicate ;</p></li>
<li><p>bam       : path to the bam file ;</p></li>
<li><p>coverage  : path to the coverage file (in rds format).</p></li>
</ul></dd>


<dt>referenceCondition</dt>
<dd><p>A <code>String</code>. Reference condition of
the RNA-Seq experiment. The reference condition specified by the
user is set to the denominator of the log2-FC.</p></dd>


<dt>locus</dt>
<dd><p>A <code>List</code>. The list contains the coordinates for the target
genomic region:</p><ul><li><p>seqid      : a <code>String</code>. Chromosome identifier
for the target genomic region ;</p></li>
<li><p>chromStart : an <code>Integer</code>. Start position for the target
genomic region ;</p></li>
<li><p>chromEnd   : an <code>Integer</code>. End position for the target
genomic region.</p></li>
</ul></dd>


<dt>stranded</dt>
<dd><p>A <code>Boolean</code>. Indicate if the reads are stranded or not.</p></dd>


<dt>strandSpecific</dt>
<dd><p>Passed on to <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>fromBam</dt>
<dd><p>A <code>Boolean</code>. Indicate if coverage profiles have to be
computed from bam files or loaded from pre-computed rds files.</p></dd>


<dt>coverageType</dt>
<dd><p>A <code>String</code>. Select how to compute the coverage profiles:</p><ul><li><p>fivePrime  : coverage profiles compute on 5' ends of reads ;</p></li>
<li><p>threePrime : coverage profiles compute on 3' ends of reads ;</p></li>
<li><p>average    : coverageFactory coverage profile compute on
average of 5' &amp; 3' ends of reads ;</p></li>
<li><p>center     : coverage profiles compute on center position of reads ;</p></li>
<li><p>fullLength : coverage profiles compute on full length reads.</p></li>
</ul></dd>


<dt>readLength</dt>
<dd><p>An <code>Integer</code>. The average length of reads from bam files.</p></dd>


<dt>isPairedEnd</dt>
<dd><p>Passed on to <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>nbThreads</dt>
<dd><p>Passed on to <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>verbose</dt>
<dd><p>A <code>Boolean</code>. Should all the operations performed be displayed ?</p></dd>


<dt>featureCountsOtherParams</dt>
<dd><p>A <code>List</code>. Other paramters passed on to
<code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code>List</code>. The list contains loaded data:</p><ul><li><p>coverage profiles  : a <code>List</code> of coverages as <code>Rle</code>;</p></li>
<li><p>log2-FC            : an <code>Rle</code> of <code>Double</code> ;</p></li>
<li><p>stranded           : see <code>loadData()</code> ;</p></li>
<li><p>locus              : the target genomic region as <code>GRanges</code>;</p></li>
<li><p>sampleInfo         : see <code>loadData()</code> ;</p></li>
<li><p>referenceCondition : see <code>loadData()</code> ;</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Create a working directory for running the example.</span></span></span>
<span class="r-in"><span><span class="va">working_directory</span> <span class="op">&lt;-</span> <span class="st">"./DIFFSEGR_TEST"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">working_directory</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Save sample information in a text file.</span></span></span>
<span class="r-in"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sample    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1"</span>, <span class="st">"wt"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  bam       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1_ChrC_71950_78500.bam"</span>, </span></span>
<span class="r-in"><span>      <span class="st">"pnp1_1_2_ChrC_71950_78500.bam"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"wt_1_ChrC_71950_78500.bam"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"wt_2_ChrC_71950_78500.bam"</span></span></span>
<span class="r-in"><span>     <span class="op">)</span>,</span></span>
<span class="r-in"><span>     <span class="kw">function</span><span class="op">(</span><span class="va">bam</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">bam</span>, package <span class="op">=</span> <span class="st">"DiffSegR"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  coverage  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">working_directory</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>, <span class="st">".rds"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">sample_info</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 sample condition replicate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_1_ChrC_71950_78500.bam pnp1_1_1    pnp1_1         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_2_ChrC_71950_78500.bam pnp1_1_2    pnp1_1         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_1_ChrC_71950_78500.bam         wt_1        wt         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_2_ChrC_71950_78500.bam         wt_2        wt         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                   bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_1_ChrC_71950_78500.bam /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/pnp1_1_1_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_2_ChrC_71950_78500.bam /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/pnp1_1_2_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_1_ChrC_71950_78500.bam         /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/wt_1_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_2_ChrC_71950_78500.bam         /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/wt_2_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                   coverage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_1_ChrC_71950_78500.bam ./DIFFSEGR_TEST/pnp1_1_1.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_2_ChrC_71950_78500.bam ./DIFFSEGR_TEST/pnp1_1_2.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_1_ChrC_71950_78500.bam         ./DIFFSEGR_TEST/wt_1.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_2_ChrC_71950_78500.bam         ./DIFFSEGR_TEST/wt_2.rds</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">sample_info</span>, </span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>, <span class="st">"sample_info.txt"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Build coverages and log2-FC.</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">loadData</span><span class="op">(</span></span></span>
<span class="r-in"><span>  sampleInfo         <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>,<span class="st">"sample_info.txt"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  locus        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqid      <span class="op">=</span> <span class="st">"ChrC"</span>, </span></span>
<span class="r-in"><span>    chromStart <span class="op">=</span> <span class="fl">71950</span>, </span></span>
<span class="r-in"><span>    chromEnd   <span class="op">=</span> <span class="fl">78500</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  referenceCondition <span class="op">=</span> <span class="st">"wt"</span>,</span></span>
<span class="r-in"><span>  stranded           <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  fromBam            <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  nbThreads          <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> building coverage profiles ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : not counted                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Read reduction : to 3' end                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid49270 ...         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 13102                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 13102                                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24249 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21098 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17531 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14066 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : not counted                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Read reduction : to 5' end                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid49270 ...         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 13102                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 13102                                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24251 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21088 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17520 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14060 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> building log2-FC per-base ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.398390531539917 secs</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$plus$pnp1_1_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 5255 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:       86        1        3        1 ...        1        1       10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$plus$pnp1_1_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 5209 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:       89        1        2        1 ...        5        2       30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.732051 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$plus$wt_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 4610 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:       46        1       15        1 ...       23        1       14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$plus$wt_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 4601 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:       57        1       26        1 ...       22        1        2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$minus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$minus$pnp1_1_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 526 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:       31        1       68        1 ...        6        1        1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$minus$pnp1_1_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 585 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:      406        1       42        1 ...        2        1        1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$minus$wt_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 463 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:      427        1       83        1 ...        4        1        2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.000000 0.414214 0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $coverages$minus$wt_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 544 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:      458        1       63        1 ...        1       18        1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values : 0.000000 0.414214 0.000000 0.414214 ... 0.414214 0.000000 0.732051</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $log2FoldChange</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $log2FoldChange$plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 6004 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:         46          1         10 ...          1          2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values :  0.0000000 -0.2500000  0.0000000 ...  -0.250000   0.000000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $log2FoldChange$minus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> numeric-Rle of length 6551 with 1239 runs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Lengths:           31            1           68 ...            1            1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Values :         0.00         0.25         0.00 ...  5.00000e-01 -3.96241e-01</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $locus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1 range and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       seqnames      ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]     ChrC 71950-78500      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $stranded</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $sampleInfo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                 sample condition replicate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_1_ChrC_71950_78500.bam pnp1_1_1    pnp1_1         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_2_ChrC_71950_78500.bam pnp1_1_2    pnp1_1         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_1_ChrC_71950_78500.bam         wt_1        wt         1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_2_ChrC_71950_78500.bam         wt_2        wt         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                                                                                   bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_1_ChrC_71950_78500.bam /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/pnp1_1_1_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_2_ChrC_71950_78500.bam /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/pnp1_1_2_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_1_ChrC_71950_78500.bam         /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/wt_1_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_2_ChrC_71950_78500.bam         /tmp/RtmpXaWfVn/temp_libpathc03c318e5828/DiffSegR/extdata/wt_2_ChrC_71950_78500.bam</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                                   coverage</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_1_ChrC_71950_78500.bam ./DIFFSEGR_TEST/pnp1_1_1.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> pnp1_1_2_ChrC_71950_78500.bam ./DIFFSEGR_TEST/pnp1_1_2.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_1_ChrC_71950_78500.bam         ./DIFFSEGR_TEST/wt_1.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> wt_2_ChrC_71950_78500.bam         ./DIFFSEGR_TEST/wt_2.rds</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $referenceCondition</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "wt"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># delete working directory </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">working_directory</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Arnaud Liehrmann, Étienne Delannoy, Benoît Castandet, Guillem Rigaill.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

