<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="In brief, segmentation() (i) identifies segments with homogeneous log2-FC
along the locus of interest and (ii) counts reads overlapping them.
In details, segmentation() (i) starts by calling fpopw::Fpop_w() on the
log2-FC profile of both strands to retreive the segments boundaries
(changepoints). The hyperparameter alpha specified by the user controls
the number of returned segments. Corresponding genomic regions are stored as
a GenomicRanges object. (ii) Finally, segmentation() calls
Rsubread::featureCounts() which assigns to the regions the mapped reads
from each replicate of each biological condition. By default a read is
allowed to be assigned to more than one region if it is found to overlap
with more than one region. Rsubread::featureCounts() ends by building a
count matrix. segmentation() ends by returning the regions and associated
count matrix as SummarizedExperiment object."><title>Summarize the differential transcription landscape — segmentation • DiffSegR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Summarize the differential transcription landscape — segmentation"><meta property="og:description" content="In brief, segmentation() (i) identifies segments with homogeneous log2-FC
along the locus of interest and (ii) counts reads overlapping them.
In details, segmentation() (i) starts by calling fpopw::Fpop_w() on the
log2-FC profile of both strands to retreive the segments boundaries
(changepoints). The hyperparameter alpha specified by the user controls
the number of returned segments. Corresponding genomic regions are stored as
a GenomicRanges object. (ii) Finally, segmentation() calls
Rsubread::featureCounts() which assigns to the regions the mapped reads
from each replicate of each biological condition. By default a read is
allowed to be assigned to more than one region if it is found to overlap
with more than one region. Rsubread::featureCounts() ends by building a
count matrix. segmentation() ends by returning the regions and associated
count matrix as SummarizedExperiment object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">DiffSegR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/introductory_tutorial.html">Introductory tutorial</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/advanced_uses.html">Advanced tutorials</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Summarize the differential transcription landscape</h1>
      
      <div class="d-none name"><code>segmentation.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>In brief, <code>segmentation()</code> (i) identifies segments with homogeneous log2-FC
along the locus of interest and (ii) counts reads overlapping them.</p>
<p>In details, <code>segmentation()</code> (i) starts by calling <code><a href="https://rdrr.io/pkg/fpopw/man/Fpop_w.html" class="external-link">fpopw::Fpop_w()</a></code> on the
log2-FC profile of both strands to retreive the segments boundaries
(changepoints). The hyperparameter <code>alpha</code> specified by the user controls
the number of returned segments. Corresponding genomic regions are stored as
a GenomicRanges object. (ii) Finally, <code>segmentation()</code> calls
<code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code> which assigns to the regions the mapped reads
from each replicate of each biological condition. By default a read is
allowed to be assigned to more than one region if it is found to overlap
with more than one region. <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code> ends by building a
count matrix. <code>segmentation()</code> ends by returning the regions and associated
count matrix as <code>SummarizedExperiment</code> object.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">segmentation</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  weightType <span class="op">=</span> <span class="st">"unweighted"</span>,</span>
<span>  modelSelectionType <span class="op">=</span> <span class="st">"yao"</span>,</span>
<span>  featureCountsType <span class="op">=</span> <span class="st">"fromBam"</span>,</span>
<span>  compressed <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  segmentNeighborhood <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  Kmax <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nbThreadsGridSearch <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  alphas <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gridSearch <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  outputDirectory <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  nbThreadsFeatureCounts <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  strandSpecific <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  read2pos <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  isPairedEnd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  featureCountsOtherParams <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>data</dt>
<dd><p>The <code>List</code> object returned by <code><a href="loadData.html">loadData()</a></code>.</p></dd>


<dt>weightType</dt>
<dd><p>A <code>String</code>. Select the type of weights associated to
the log2-FC per-base:</p><ul><li><p>unweighted   : all observations have the same weight (=1) ;</p></li>
<li><p>zeroInflated : low counts have less weight.</p></li>
</ul></dd>


<dt>modelSelectionType</dt>
<dd><p>A <code>String</code>. Select the penalty used by FPOP:</p><ul><li><p>yao : Yao's penalty <code>alpha*sigma^2*log(n)</code>.</p></li>
</ul></dd>


<dt>featureCountsType</dt>
<dd><p>A <code>String</code>. Select how to summarize counts:</p><ul><li><p>fromBam      : from bam files using <a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts</a> ;</p></li>
<li><p>fromCoverage : from coverage profiles.</p></li>
</ul></dd>


<dt>compressed</dt>
<dd><p>A <code>Boolean</code>. Indicate if the observations have to be
compressed (does not change the segmentation results and decreases
the running time).</p></dd>


<dt>alpha</dt>
<dd><p>A <code>Double</code>. Segmentation hyperparameter used in Yao's penalty:
<code>alpha*sigma^2*log(n)</code>. The number of changepoints returned by
<code><a href="https://rdrr.io/pkg/fpopw/man/Fpop_w.html" class="external-link">fpopw::Fpop_w()</a></code> is a decreasing function of <code>alpha</code>.</p></dd>


<dt>segmentNeighborhood</dt>
<dd><p>A <code>Boolean</code>. Indicate the weighted pDPA algorithm
(Rigaill 2010 and 2015) has to be used to explore the space of segmentations.</p></dd>


<dt>Kmax</dt>
<dd><p>An <code>Integer</code>. Segmentations with 1 to Kmax segments are recovered
using segment neighborhood.</p></dd>


<dt>verbose</dt>
<dd><p>A <code>Boolean</code>. Should all the operations performed be
displayed ?</p></dd>


<dt>nbThreadsGridSearch</dt>
<dd><p>An <code>Integer</code>. Number of threads used by the grid
search procedure.</p></dd>


<dt>alphas</dt>
<dd><p>A vector of <code>Double</code>. A series of alphas used by the grid
search procedure.</p></dd>


<dt>gridSearch</dt>
<dd><p>A <code>Boolean</code>. Indicate if a grid search has to be used
to explore the space of segmentations.</p></dd>


<dt>outputDirectory</dt>
<dd><p>A <code>String</code>. Path to the output directory.</p></dd>


<dt>nbThreadsFeatureCounts</dt>
<dd><p>An <code>Integer</code>. Passed on to
<code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>strandSpecific</dt>
<dd><p>An <code>Integer</code>. Passed on to <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>read2pos</dt>
<dd><p>An <code>Integer</code>. Passed on to <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>isPairedEnd</dt>
<dd><p>Passed on to <code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>


<dt>featureCountsOtherParams</dt>
<dd><p>A <code>List</code>. Other paramters passed on to
<code><a href="https://rdrr.io/pkg/Rsubread/man/featureCounts.html" class="external-link">Rsubread::featureCounts()</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <code>SummarizedExperiment</code> object that contains region boundaries and
associated count matrix.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create a working directory for running the example.</span></span></span>
<span class="r-in"><span><span class="va">working_directory</span> <span class="op">&lt;-</span> <span class="st">"./DIFFSEGR_TEST"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">working_directory</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Save sample information in a text file.</span></span></span>
<span class="r-in"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sample    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  condition <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1"</span>, <span class="st">"wt"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  replicate <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  bam       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1_ChrC_71950_78500.bam"</span>, </span></span>
<span class="r-in"><span>      <span class="st">"pnp1_1_2_ChrC_71950_78500.bam"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"wt_1_ChrC_71950_78500.bam"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"wt_2_ChrC_71950_78500.bam"</span></span></span>
<span class="r-in"><span>     <span class="op">)</span>,</span></span>
<span class="r-in"><span>     <span class="kw">function</span><span class="op">(</span><span class="va">bam</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">bam</span>, package <span class="op">=</span> <span class="st">"DiffSegR"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  coverage  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">working_directory</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pnp1_1_1"</span>, <span class="st">"pnp1_1_2"</span>, <span class="st">"wt_1"</span>, <span class="st">"wt_2"</span><span class="op">)</span>, <span class="st">".rds"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">sample_info</span>, </span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>, <span class="st">"sample_info.txt"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Build coverages and log2-FC.</span></span></span>
<span class="r-in"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="loadData.html">loadData</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  sampleInfo         <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">working_directory</span>,<span class="st">"sample_info.txt"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  locus        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    seqid      <span class="op">=</span> <span class="st">"ChrC"</span>, </span></span>
<span class="r-in"><span>    chromStart <span class="op">=</span> <span class="fl">71950</span>, </span></span>
<span class="r-in"><span>    chromEnd   <span class="op">=</span> <span class="fl">78500</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  referenceCondition <span class="op">=</span> <span class="st">"wt"</span>,</span></span>
<span class="r-in"><span>  stranded           <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  fromBam            <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  nbThreads          <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> building coverage profiles ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : not counted                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Read reduction : to 3' end                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid1296978 ...       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 13102                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 13102                                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24249 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21098 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17531 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14066 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : not counted                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Read reduction : to 5' end                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid1296978 ...       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 13102                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 13102                                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24251 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21088 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17520 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14060 (99.9%)                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> building log2-FC per-base ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.473139047622681 secs</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Summarize the differential landscape.</span></span></span>
<span class="r-in"><span><span class="va">SExp</span> <span class="op">&lt;-</span> <span class="fu">segmentation</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data                   <span class="op">=</span> <span class="va">data</span>, </span></span>
<span class="r-in"><span>  nbThreadsFeatureCounts <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  outputDirectory        <span class="op">=</span> <span class="va">working_directory</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> changepoint detection on ChrC 71950 78500 plus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; model selection ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; report available in ./DIFFSEGR_TEST/all_models_plus.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> changepoint detection on ChrC 71950 78500 minus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; model selection ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   &gt; report available in ./DIFFSEGR_TEST/all_models_minus.pdf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.156652212142944 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> counting ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //========================== featureCounts setting ===========================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Input files : 4 BAM files                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_1_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           pnp1_1_2_ChrC_71950_78500.bam                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_1_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           wt_2_ChrC_71950_78500.bam                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Paired-end : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||        Count read pairs : no                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Annotation : R data.frame                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Dir for temp files : .                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Threads : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                   Level : meta-feature level                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||      Multimapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Multi-overlapping reads : counted                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   Min overlapping bases : 1                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load annotation file .Rsubread_UserProvidedAnnotation_pid1296978 ...       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Features : 69                                                           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Meta-features : 69                                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Chromosomes/contigs : 1                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_1_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 24260                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 24260 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file pnp1_1_2_ChrC_71950_78500.bam...                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 21100                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 21100 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_1_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 17533                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 17533 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Process BAM file wt_2_ChrC_71950_78500.bam...                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Strand specific : stranded                                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Single-end reads are included.                                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Total alignments : 14070                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Successfully assigned alignments : 14070 (100.0%)                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    Running time : 0.00 minutes                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the final count table.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Write the read assignment summary.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> finished in 0.125180721282959 secs</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># In genomic order, first to fifth regions and ...</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu">mcols</span><span class="op">(</span><span class="va">SExp</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 5 rows and 4 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       modelStart  modelEnd modelMean             featureId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                        &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;           &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus          1       306 0.0257894 ChrC_71950_72255_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus        307       453 0.6352520 ChrC_72256_72402_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus        454      2089 0.2826332 ChrC_72403_74038_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus       2090      2356 0.0910074 ChrC_74039_74305_plus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus       2357      2892 0.3207993 ChrC_74306_74841_plus</span>
<span class="r-in"><span><span class="co"># ... associated counts.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">SExp</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       pnp1_1_1 pnp1_1_2  wt_1 wt_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_71950_72255_plus      143      139    84   84</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72256_72402_plus      649      665   326  293</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_72403_74038_plus    13203    11731 10849 8366</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74039_74305_plus      155      132    98  104</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ChrC_74306_74841_plus     2337     1996  1589 1218</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># delete working directory </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">working_directory</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Arnaud Liehrmann, Benoît Castandet, Guillem Rigaill.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

